qutebrowser/config/config.py

```diff
@@ -640,23 +640,32 @@ class StyleSheetObserver(QObject):
     Attributes:
         _obj: The object to observe.
         _stylesheet: The stylesheet template to use.
+        _options: The config options that the stylesheet uses. When it's not
+                  necessary to listen for config changes, this attribute may be
+                  missing.
     """
 
     def __init__(self, obj: QObject,
-                 stylesheet: typing.Optional[str],
-                 update: bool) -> None:
+                 stylesheet: Optional[str], update: bool) -> None:
         super().__init__()
         self._obj = obj
         self._update = update
 
         # We only need to hang around if we are asked to update.
-        if self._update:
+        if update:
             self.setParent(self._obj)
         if stylesheet is None:
             self._stylesheet = obj.STYLESHEET  # type: str
         else:
             self._stylesheet = stylesheet
 
+        if update:
+            self._options = jinja.template_config_variables(
+                self._stylesheet)  # type: FrozenSet[str]
+
+            # assert that all of the options exists
+            assert all(instance.get_opt(option) for option in self._options)
```

Comment:
This seems a bit weird to me, since it suggests that `get_opt` would return a boolean - but we're actually just checking whether there is no exception, right?

I'd prefer:

```python
for option in self._options:
    instance.get_opt(option)
```

instead. Also, shouldn't this be part of `jinja.template_config_variables` instead? It seems like a good sanity check to have there, and then you can probably also benefit from the caching and not do this multiple times for the same stylesheet.

Reply:
I find it a bit weird too, but I prefer stuffing sanity check into assertions.

Yes, that may be a good idea, the function is specialized for config already.

Reply:
There seems to be quite some (currently 3) of `get_opt` currently used to check existence. Perhaps we should convert them to `assert has_opt(name)`. Also avoid having to read some comment.

Reply:
Is this a good idea?

https://github.com/user202729/qutebrowser/tree/assert-config-has-opt

Reply:
Sounds good!